modules/uhttp/http.go

```diff
@@ -189,14 +190,19 @@ func (m *Module) Start(ready chan<- struct{}) <-chan error {
 	}
 	m.listenMu.Lock()
 	m.listener = listener
+	m.srv = &http.Server{
+		Handler: m.mux,
+	}
 	m.listenMu.Unlock()
 
 	go func() {
 		listener := m.accessListener()
 		ready <- struct{}{}
-		ret <- nil
-		if err := http.Serve(listener, m.mux); err != nil {
+		if err := m.srv.Serve(listener); err != nil {
 			m.log.Error("HTTP Serve error", "error", err)
+			ret <- err
```

Comment:
I tried, but it's really hard to induce an error now. I could artificially close the `listener` but it wouldn't really be testing a code path that would be hit in production, since we don't expose a mutator to the listener.